bears/css/StyleLintBear.py

```diff
@@ -27,139 +29,569 @@ class StyleLintBear:
     CAN_DETECT = {'Syntax', 'Unused Code', 'Formatting'}
 
     @staticmethod
-    def generate_config(filename, file):
-        # Use standard stylelint rules
+    def generate_config(
+        filename, file,
+        at_rule_empty_line_before: str = None,
+        at_rule_name_case: str = 'lower',
+        at_rule_name_space_after: str = 'always-single-line',
+        at_rule_semicolon_newline_after: str = 'always',
+        block_closing_brace_empty_line_before: str = 'never',
+        block_closing_brace_newline_after: str = 'always',
+        block_closing_brace_newline_before: str = 'always-multi-line',
+        block_closing_brace_space_before: str = 'always-single-line',
+        block_no_empty: bool = True,
+        block_opening_brace_newline_after: str = 'always-multi-line',
+        block_opening_brace_space_after: str = 'always-single-line',
+        block_opening_brace_space_before: str = 'always',
+        color_hex_case: str = 'lower',
+        color_hex_length: str = 'short',
+        color_no_invalid_hex: bool = True,
+        comment_empty_line_before: str = None,
+        comment_no_empty: bool = True,
+        comment_whitespace_inside: str = 'always',
+        custom_property_empty_line_before: str = None,
+        declaration_bang_space_after: str = 'never',
+        declaration_bang_space_before: str = 'always',
+        declaration_block_no_duplicate_properties: bool = None,
+        declaration_block_no_redundant_longhand_properties: bool = True,
+        declaration_block_no_shorthand_property_overrides: bool = True,
+        declaration_block_semicolon_newline_after: str = 'always-multi-line',
+        declaration_block_semicolon_space_after: str = 'always-single-line',
+        declaration_block_semicolon_space_before: str = 'never',
+        declaration_block_single_line_max_declarations: int = 1,
+        declaration_block_trailing_semicolon: str = 'always',
+        declaration_colon_newline_after: str = 'always-multi-line',
+        declaration_colon_space_after: str = 'always-single-line',
+        declaration_colon_space_before: str = 'never',
+        declaration_empty_line_before: str = None,
+        font_family_no_duplicate_names: bool = True,
+        function_calc_no_unspaced_operator: bool = True,
+        function_comma_newline_after: str = 'always-multi-line',
+        function_comma_space_after: str = 'always-single-line',
+        function_comma_space_before: str = 'never',
+        function_linear_gradient_no_nonstandard_direction: bool = True,
+        function_max_empty_lines: int = 0,
+        function_name_case: str = 'lower',
+        function_parentheses_newline_inside: str = 'always-multi-line',
+        function_parentheses_space_inside: str = 'never-single-line',
+        function_whitespace_after: str = 'always',
+        indentation: Union[int, str] = 2,
```

Comment:
im not sure our setting system supports Union.  If it doesnt, a fix is needed in coala_utils or coala core.

Also, 'typing' doesnt existing on Python 3.4 core, so you need to add a dependency on pypi package `typing` (ideally only on py34, but that may also turn out to be difficult, so maybe we defer that atm).